android/tests_backend/window.py

```diff
@@ -18,14 +22,39 @@ def __init__(self, app, window):
     async def wait_for_window(
         self,
         message,
-        minimize=False,
-        full_screen=False,
-        state_switch_not_from_normal=False,
+        state=None,
     ):
-        await self.redraw(
-            message,
-            delay=(0.5 if (full_screen or state_switch_not_from_normal) else 0.1),
-        )
+        await self.redraw(message, delay=0.1)
+        if state:
+            timeout = 5
+            polling_interval = 0.1
+            exception = None
+            loop = asyncio.get_running_loop()
+            start_time = loop.time()
+            while (loop.time() - start_time) < timeout:
+                try:
+                    assert self.instantaneous_state == state
+                    if state in {WindowState.FULLSCREEN, WindowState.PRESENTATION}:
+                        # Add a slight delay to ensure window properties like
+                        # `content_size` are updated according to the new state.
+                        await self.redraw(delay=0.1)
+                    return
```

Comment:
I would have added another assertion of `assert content_size > initial_content_size` on the android `wait_for_window()` probe method, like I have done on the cocoa backend for `assert self.window._impl._pending_state_transition is None`. 

But, for `content_size`, we donot have access to the `initial_content_size` on the Android probe, so the content_size assertion cannot be done on the polling mechanism. Therefore, I have added a slight delay in order to ensure that the window properties like `content_size` are updated according to the new state. 

I had also thought of adding a flag to the Android's `LayoutListener`, but I realized that there would no suitable place to reset the flag, and overall would cause more code churning.

Reply:
Would this be an acceptable solution?

Reply:
Yes - it's not ideal (as you've noted ideally we'd be polling for an observable condition), but it's at least as good as we had before.